functions/Grant-YouTubeOauth.ps1

```diff
@@ -43,16 +48,10 @@ function Grant-YoutubeOauth {
     'https://www.googleapis.com/auth/youtube.force-ssl'
     'https://www.googleapis.com/auth/youtube.readonly'
   )
-  $Uri = 'https://accounts.google.com/o/oauth2/v2/auth?include_granted_scopes=true&response_type=token&client_id={0}&redirect_uri={1}&scope={3}&state={2}' -f $Client.client_id, $RedirectUri, (New-Guid).Guid, ($ScopeList -join ' ')
+  $scopesURI = [System.Web.HttpUtility]::UrlEncode($ScopeList -join ' ')
+  $Uri = 'https://accounts.google.com/o/oauth2/v2/auth?include_granted_scopes=true&response_type=token&client_id={0}&redirect_uri={1}&scope={2}&state={2}' -f $Client.client_id, $RedirectUri, ($scopesURI), (New-Guid).Guid
 
   $Browser = $BrowserCommand ? $BrowserCommand : (Find-Browser)
   Write-Verbose -Message ('Browser command line is: ' -f $Browser)
-  Start-Process -FilePath $Browser -ArgumentList ('"{0}"' -f $Uri) -Wait
-
-  if (!$IsMacOS) {
-    Stop-Job -Name $JobName
-  }
-  else {
-    Write-Warning -Message 'Please manually stop the Background Job running the Pode web server after completing authentication. Cannot wait for browser process to complete on MacOS.'
-  }
-}
+  Start-Process -FilePath $Browser -ArgumentList ('"{0}"' -f $Uri)
```

Comment:
I remember putting this logic in for MacOS. Did you test this on MacOS?

Reply:
no longer needed because the job can always complete. see above.
I do not have a MAC. Did not test.